bundles/EcommerceFrameworkBundle/OrderManager/Order/Agent.php

```diff
@@ -163,15 +163,35 @@ public function itemChangeAmount(OrderItem $item, $amount)
         $oldAmount = $item->getAmount();
         $note->addData('amount.old', 'text', $oldAmount);
         $note->addData('amount.new', 'text', $amount);
-
+        $oldTotalPrice = $item->getTotalPrice();
+        $unitPrice = $oldTotalPrice/$oldAmount;
+        $newTotalPrice =$unitPrice*$amount;
         // change
         $item->setAmount($amount);
-
+        $item->setTotalPrice( $newTotalPrice);
         // save
         $item->save();
         $note->save();
-
+        $this->updateOrderTotal($item,$oldTotalPrice,$newTotalPrice);
         return $note;
+
+    }
+
+    
+    public function updateOrderTotal($item,$oldTotalPrice,$newTotalPrice)
+    {  
+       
+        $order = $item->getParent();
+        $subTotalNetPrice = $order->getSubTotalNetPrice();
+        $subTotalPrice = $order->getSubTotalPrice();
+        $totalNetPrice = $order->getTotalNetPrice();
+        $totalPrice = $order->getTotalPrice();
+        $replacedvalue  = $newTotalPrice -$oldTotalPrice;
+        $order->setSubTotalNetPrice($subTotalNetPrice +$replacedvalue);
+        $order->setSubTotalPrice($subTotalPrice+$replacedvalue);
+        $order->setTotalNetPrice($totalNetPrice+$replacedvalue);
```

Comment:
that calculation is wrong ... cannot reduce gross and net price with same amount. also need to update the tax information.